Makefile

```diff
@@ -93,129 +92,49 @@ release: certs docker-release
 	@docker build -f scripts/dockerfiles/Dockerfile.release -t "amazon/amazon-ecs-agent:latest" .
 	@echo "Built Docker image \"amazon/amazon-ecs-agent:latest\""
 
-gogenerate:
-	go generate -x $(shell go list ./agent/...)
-	$(MAKE) goimports
-
 # We need to bundle certificates with our scratch-based container
 certs: misc/certs/ca-certificates.crt
 misc/certs/ca-certificates.crt:
 	docker build -t "amazon/amazon-ecs-agent-cert-source:make" misc/certs/
 	docker run "amazon/amazon-ecs-agent-cert-source:make" cat /etc/ssl/certs/ca-certificates.crt > misc/certs/ca-certificates.crt
 
-ifeq (${BUILD_PLATFORM},aarch64)
-test::
-	go test -tags unit -timeout=30s -v -cover $(shell go list ./agent/... | grep -v /vendor/)
-else
-test::
-	go test -race -tags unit -timeout=30s -v -cover $(shell go list ./agent/... | grep -v /vendor/)
-endif
-
-ifeq (${BUILD_PLATFORM},aarch64)
-test-silent::
-	go test -tags unit -timeout=30s -cover $(shell go list ./agent/... | grep -v /vendor/)
-else
-test-silent::
-	go test -race -tags unit -timeout=30s -cover $(shell go list ./agent/... | grep -v /vendor/)
-endif
-
-benchmark-test:
-	go test -run=XX -bench=. $(shell go list ./agent/... | grep -v /vendor/)
+gogenerate:
+	go generate -x ./agent/...
+	$(MAKE) goimports
 
-define dockerbuild
-	docker run \
-		--net none \
-		--user "$(USERID)" \
-		--volume "$(PWD)/out:/out" \
-		--volume "$(PWD):/go/src/github.com/aws/amazon-ecs-agent" \
-		--rm \
-		$(BUILDER_IMAGE) \
-		$(1)
-endef
+# 'go' may not be on the $PATH for sudo tests
+GO_EXECUTABLE=$(shell command -v go 2> /dev/null)
 
-define win-cgo-dockerbuild
-	docker run --net=none \
-		--user "$(USERID)" \
-		--volume "$(PWD)/out:/out" \
-		--volume "$(PWD):/go/src/github.com/aws/amazon-ecs-agent" \
-		--env "GOOS=windows" \
-		--env "CGO_ENABLED=1" \
-		--env "CC=x86_64-w64-mingw32-gcc" \
-		--rm \
-		$(BUILDER_IMAGE) \
-		$(1)
-endef
+# VERBOSE includes the options that make the test opt loud
+VERBOSE=-v -cover
 
-# TODO: use `go list -f` to target the test files more directly
-ALL_GO_FILES = $(shell find . -name "*.go" -print | tr "\n" " ")
+# -count=1 prevents the build cache from breaking integ tests
```

Comment:
This wording is less clear than what you said in the message above `Use -count=1 when running any of the tests to prevent caching from causing goofy test results`.  Can you tweak this to be super-clear that count=1 prevents caching.

Reply:
Yep, good point